services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankealueService.kt

```diff
@@ -0,0 +1,157 @@
+package fi.hel.haitaton.hanke
+
+import fi.hel.haitaton.hanke.application.CableReportApplicationData
+import fi.hel.haitaton.hanke.domain.Hankealue
+import fi.hel.haitaton.hanke.domain.HasId
+import fi.hel.haitaton.hanke.domain.NewGeometriat
+import fi.hel.haitaton.hanke.domain.NewHankealue
+import fi.hel.haitaton.hanke.domain.SavedHankealue
+import fi.hel.haitaton.hanke.geometria.Geometriat
+import fi.hel.haitaton.hanke.geometria.GeometriatService
+import fi.hel.haitaton.hanke.tormaystarkastelu.TormaystarkasteluLaskentaService
+import fi.hel.haitaton.hanke.tormaystarkastelu.TormaystarkasteluTulosEntity
+import org.geojson.Feature
+import org.geojson.FeatureCollection
+import org.springframework.stereotype.Service
+
+/*
+ * Helper for mapping and sorting data to existing collections.
+ *
+ * Example usage:
+ * mergeDataInto(hanke.alueet, entity.listOfHankeAlueet) { source, target ->
+ *    copyNonNullHankealueFieldsToEntity(hanke, source, target)
+ * }
+ * - Transforms data from hanke.alueet into entity.listOfHankeAlueet
+ * - Does the transformation with the last lambda
+ *
+ * Source list is not modified. Target container is sorted by order of source container.
+ *
+ * converter takes additional parameter for existing Target object in case it exists in target collection.
+ *
+ */
+private fun <Source : HasId<Int>, Target : HasId<Int>> mergeDataInto(
```

Comment:
If HankealueService is the only class that needs `mergeDataInto` then could it be a member function? Otherwise, it might be better to put it into `Utils.kt` to keep it as a general utility method?

Reply:
I thought I would try to make this more useful and less weird, but I don't know if I succeeded.

Looking at this led me to some refactoring in commit 49b3486:
- I made the type parameter in HasId exact, so the implementation
  specifies if the id property is nullable or not.
- I changed the id property in HankealueEntity to be non-nullable.
- I changed the mergeDataInto function to accept only target types with
  non-nullable IDs. This removes the need to handle cases where the
  target ID would be null. Move the function to Utils.kt and add tests
  for it.
- Upgrade ktfmt used by the Spotless formatter. The old version crashed
  when using Kotlin's definitely non-nullable types feature (X & Any).